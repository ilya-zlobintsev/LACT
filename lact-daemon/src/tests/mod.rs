use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
