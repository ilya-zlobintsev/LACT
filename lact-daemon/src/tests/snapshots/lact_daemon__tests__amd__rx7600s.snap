---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": null,
  "info": {
    "driver": "amdgpu",
    "drm_info": {
      "vram_clock_ratio": 1.0
    },
    "flags": [
      "DumpableVBios",
      "ConfigurableFanControl"
    ],
    "link_info": {
      "current_speed": "16.0 GT/s PCIe",
      "current_width": "8"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 33 [Radeon RX 7600/7600 XT/7600M XT/7600S/7700S / PRO W7600]",
        "model_id": "7480",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model_id": "232D",
        "vendor": "ASUSTeK Computer Inc.",
        "vendor_id": "1043"
      }
    },
    "vbios_version": "113-BRT112125.001",
    "vulkan_instances": []
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 33 [Radeon RX 7600/7600 XT/7600M XT/7600S/7700S / PRO W7600]",
      "model_id": "7480",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model_id": "232D",
      "vendor": "ASUSTeK Computer Inc.",
      "vendor_id": "1043"
    }
  },
  "power_profile_modes": {
    "active": 4,
    "modes": {
      "0": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              0,
              0,
              1,
              4,
              0,
              800,
              0,
              0,
              0,
              3,
              1,
              0,
              0
            ]
          }
        ],
        "name": "BOOTUP_DEFAULT"
      },
      "1": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              2,
              0,
              1,
              4,
              0,
              650,
              0,
              0,
              0,
              1,
              1,
              0,
              0
            ]
          }
        ],
        "name": "3D_FULL_SCREEN"
      },
      "2": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              0,
              0,
              1,
              3,
              0,
              0,
              0,
              0,
              0,
              1,
              1,
              0,
              0
            ]
          }
        ],
        "name": "POWER_SAVING"
      },
      "3": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              0,
              0,
              1,
              4,
              0,
              0,
              0,
              0,
              0,
              1,
              1,
              0,
              0
            ]
          }
        ],
        "name": "VIDEO"
      },
      "4": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              1,
              0,
              4,
              1,
              1000,
              0,
              0,
              0,
              0,
              1,
              1,
              0,
              0
            ]
          }
        ],
        "name": "VR"
      },
      "5": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              1,
              0,
              4,
              1,
              1000,
              0,
              0,
              0,
              0,
              1,
              1,
              0,
              0
            ]
          }
        ],
        "name": "COMPUTE"
      },
      "6": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              0,
              0,
              1,
              4,
              0,
              800,
              0,
              0,
              0,
              3,
              1,
              0,
              0
            ]
          }
        ],
        "name": "CUSTOM"
      },
      "7": {
        "components": [
          {
            "clock_type": null,
            "values": [
              0,
              2,
              0,
              1,
              4,
              0,
              650,
              0,
              0,
              0,
              3,
              1,
              0,
              0
            ]
          }
        ],
        "name": "WINDOW_3D"
      }
    },
    "value_names": [
      "Gfx_ActiveHystLimit",
      "Gfx_IdleHystLimit",
      "Gfx_FPS",
      "Gfx_MinActiveFreqType",
      "Gfx_BoosterFreqType",
      "Gfx_MinActiveFreq",
      "Gfx_BoosterFreq",
      "Fclk_ActiveHystLimit",
      "Fclk_IdleHystLimit",
      "Fclk_FPS",
      "Fclk_MinActiveFreqType",
      "Fclk_BoosterFreqType",
      "Fclk_MinActiveFreq",
      "Fclk_BoosterFreq"
    ]
  },
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 255
      },
      {
        "enabled": true,
        "index": 1,
        "value": 0
      },
      {
        "enabled": true,
        "index": 2,
        "value": 1841
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 96
      },
      {
        "enabled": true,
        "index": 1,
        "value": 456
      },
      {
        "enabled": true,
        "index": 2,
        "value": 772
      },
      {
        "enabled": true,
        "index": 3,
        "value": 1000
      }
    ]
  },
  "stats": {
    "busy_percent": 0,
    "clockspeed": {
      "gpu_clockspeed": 0,
      "vram_clockspeed": 96
    },
    "core_power_state": 1,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 76,
      "pwm_max": 255,
      "pwm_min": 0,
      "speed_current": 0,
      "speed_max": 4900,
      "speed_min": 0
    },
    "memory_power_state": 0,
    "performance_level": "manual",
    "power": {
      "average": 1.0,
      "cap_current": 95.0,
      "cap_default": 95.0,
      "cap_max": 95.0,
      "cap_min": 95.0
    },
    "temps": {
      "edge": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 53.0
      },
      "junction": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 54.0
      },
      "mem": {
        "crit": 105.0,
        "crit_hyst": -273.1499938964844,
        "current": 60.0
      }
    },
    "voltage": {
      "gpu": 0
    },
    "vram": {}
  }
}
